### PR TITLE
Add label for pixel digitizer in SiPixelQualityESProducer

### PR DESCRIPTION
**Pull request by tsusa**
This PR introduces the label for the `SiPixelQualityRcd` in order to treat the damaged pixels behind broken DCDC converters the same way in data and in simulation. The PR was explained at the RECO meeting https://indico.cern.ch/event/787904/contributions/3288669/subcontributions/273135/attachments/1785559/2906928/tanjaReco_25012019.pdf.
The feature is switched off by default. In order to switch it on:
 - the new GT  containing `SiPixelQuality` with the label "forDigitizer" is needed 
 - the lines L108-109 in SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py should be uncommented 
The Era mechanism is used to activate the feature only for >= 2018

### CODE REVIEW

**Comment by tsusa** [comment]
@tvami @leaca

**Comment by fabiocos** [comment]
please test

File: SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py [hunk 1]
```diff
@@ -103,6 +104,10 @@ def _modifyPixelDigitizerForPhase1Pixel( digitizer ) :
 from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
 phase1Pixel.toModify( SiPixelSimBlock, func=_modifyPixelDigitizerForPhase1Pixel )
 
+# To use the label for 2018 data, uncomment next two lines, new GT needed
+#from Configuration.Eras.Modifier_run2_SiPixel_2018_cff import run2_SiPixel_2018
+#run2_SiPixel_2018.toModify(SiPixelSimBlock, SiPixelQualityLabel = 'forDigitizer',)
+
 from Configuration.ProcessModifiers.premix_stage1_cff import premix_stage1
 premix_stage1.toModify(SiPixelSimBlock,
```
Comment (Martin-Grunewald) [inline SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py]: I see the two lines are already uncommented. So the feature is switched on, in contrast what is written in the PR description?
Likely the reason why the AddOn tests fail, given the error messages there.

Comment (tsusa) [reply SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py]: Hi @Martin-Grunewald, those two lines are still commented   
# To use the label for 2018 data, uncomment next two lines, new GT needed
#from Configuration.Eras.Modifier_run2_SiPixel_2018_cff import run2_SiPixel_2018
#run2_SiPixel_2018.toModify(SiPixelSimBlock, SiPixelQualityLabel = 'forDigitizer',)

Comment (Martin-Grunewald) [reply SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py]: Well, it is switched on somewhere:
```
----- Begin Fatal Exception 08-Feb-2019 13:14:16 CET-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Processing  stream begin Run run: 1 stream: 2
   [1] Calling method for module EventSetupRecordDataGetter/'hltGetConditions'
   [2] Using EventSetup component SiPixelQualityESProducer/'siPixelQualityESProducer' to make data SiPixelQuality/'forDigitizer' in record SiPixelQualityRcd
Exception Message:
No data of type "SiPixelQuality" with label "forDigitizer" in record "SiPixelQualityFromDbRcd"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------
```

**Comment by tsusa** [comment]
In all candidate GTs, the unlabelled and labelled SiPixelQuality payloads are the same, therefore no difference is expected in the comparison plots. Can you please start the tests?

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@tocheng FYI:
since the queues were updated from the time the last GTs were cut, the new candidates differ a bit from the old Global Tags, mostly because of JECs, but in the post-LS2 case also becasue of Tracker Alignment. This might generate some spurious differences in the comparisons.

```
Record                   Label         pro::103X_upgrade2018_design_v4 Tag  pro::103X_upgrade2018_design_Candidate_2019_02_08_17_44_35 Tag  
-----------------------  ------------  -----------------------------------  --------------------------------------------------------------  
JetResolutionRcd         AK4PFPuppi    -                                    JR_Fall17_V3_102X_MC_PhiResolution_AK4PFPuppi                   
JetResolutionRcd         AK8PF         -                                    JR_Fall17_V3_102X_MC_PhiResolution_AK8PF                        
JetResolutionRcd         AK8PFPuppi    -                                    JR_Fall17_V3_102X_MC_PtResolution_AK8PFPuppi                    
JetResolutionRcd         AK8PFchs      -                                    JR_Fall17_V3_102X_MC_PtResolution_AK8PFchs                      
SiPixelQualityFromDbRcd  forDigitizer  -                                    SiPixelQuality_phase1_ideal                                     

Record                       Label           pro::103X_upgrade2018_realistic_v8 Tag                               pro::103X_upgrade2018_realistic_Candidate_2019_02_08_16_57_06 Tag         
---------------------------  --------------  -------------------------------------------------------------------  ------------------------------------------------------------------------  
JetCorrectionsRecord         AK4Calo         JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4Calo     JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4Calo     
JetCorrectionsRecord         AK4JPT          JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4JPT      JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4JPT      
JetCorrectionsRecord         AK4PF           JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PF       JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PF       
JetCorrectionsRecord         AK4PFPuppi      JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PFPuppi  JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PFPuppi  
JetCorrectionsRecord         AK4PFchs        JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PFchs    JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PFchs    
JetCorrectionsRecord         AK8PF           JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PF       JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PF       
JetCorrectionsRecord         AK8PFPuppi      JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PFPuppi  JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PFPuppi  
JetCorrectionsRecord         AK8PFchs        JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PFchs    JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PFchs    
JetResolutionRcd             AK4PFPuppi_eta  -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PFPuppi                             
JetResolutionRcd             AK4PFPuppi_phi  JR_Spring16_25nsV6_MC_PhiResolution_AK4PFPuppi                       JR_Fall17_V3_102X_MC_PhiResolution_AK4PFPuppi                             
JetResolutionRcd             AK4PFPuppi_pt   JR_Spring16_25nsV6_MC_PtResolution_AK4PFPuppi                        JR_Fall17_V3_102X_MC_PtResolution_AK4PFPuppi                              
JetResolutionRcd             AK4PF_eta       -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PF                                  
JetResolutionRcd             AK4PF_phi       JR_Spring16_25nsV6_MC_PhiResolution_AK4PF                            JR_Fall17_V3_102X_MC_PhiResolution_AK4PF                                  
JetResolutionRcd             AK4PF_pt        JR_Spring16_25nsV6_MC_PtResolution_AK4PF                             JR_Fall17_V3_102X_MC_PtResolution_AK4PF                                   
JetResolutionRcd             AK4PFchs_eta    -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PFchs                               
JetResolutionRcd             AK4PFchs_phi    JR_Spring16_25nsV6_MC_PhiResolution_AK4PFchs                         JR_Fall17_V3_102X_MC_PhiResolution_AK4PFchs                               
JetResolutionRcd             AK4PFchs_pt     JR_Spring16_25nsV6_MC_PtResolution_AK4PFchs                          JR_Fall17_V3_102X_MC_PtResolution_AK4PFchs                                
JetResolutionRcd             AK8PFPuppi_eta  -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PFPuppi                             
JetResolutionRcd             AK8PFPuppi_phi  JR_Spring16_25nsV6_MC_PhiResolution_AK8PFPuppi                       JR_Fall17_V3_102X_MC_PhiResolution_AK8PFPuppi                             
JetResolutionRcd             AK8PFPuppi_pt   JR_Spring16_25nsV6_MC_PtResolution_AK8PFPuppi                        JR_Fall17_V3_102X_MC_PtResolution_AK8PFPuppi                              
JetResolutionRcd             AK8PF_eta       -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PF                                  
JetResolutionRcd             AK8PF_phi       JR_Spring16_25nsV6_MC_PhiResolution_AK8PF                            JR_Fall17_V3_102X_MC_PhiResolution_AK8PF                                  
JetResolutionRcd             AK8PF_pt        JR_Spring16_25nsV6_MC_PtResolution_AK8PF                             JR_Fall17_V3_102X_MC_PtResolution_AK8PF                                   
JetResolutionRcd             AK8PFchs_eta    -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PFchs                               
JetResolutionRcd             AK8PFchs_phi    JR_Spring16_25nsV6_MC_PhiResolution_AK8PFchs                         JR_Fall17_V3_102X_MC_PhiResolution_AK8PFchs                               
JetResolutionRcd             AK8PFchs_pt     JR_Spring16_25nsV6_MC_PtResolution_AK8PFchs                          JR_Fall17_V3_102X_MC_PtResolution_AK8PFchs                                
JetResolutionScaleFactorRcd  AK4PF           JR_Spring16_25nsV6_MC_SF_AK4PF                                       JR_Fall17_V3_102X_MC_SF_AK4PF                                             
JetResolutionScaleFactorRcd  AK4PFPuppi      JR_Spring16_25nsV6_MC_SF_AK4PFPuppi                                  JR_Fall17_V3_102X_MC_SF_AK4PFPuppi                                        
JetResolutionScaleFactorRcd  AK4PFchs        JR_Spring16_25nsV6_MC_SF_AK4PFchs                                    JR_Fall17_V3_102X_MC_SF_AK4PFchs                                          
JetResolutionScaleFactorRcd  AK8PF           JR_Spring16_25nsV6_MC_SF_AK8PF                                       JR_Fall17_V3_102X_MC_SF_AK8PF                                             
JetResolutionScaleFactorRcd  AK8PFPuppi      JR_Spring16_25nsV6_MC_SF_AK8PFPuppi                                  JR_Fall17_V3_102X_MC_SF_AK8PFPuppi                                        
JetResolutionScaleFactorRcd  AK8PFchs        JR_Spring16_25nsV6_MC_SF_AK8PFchs                                    JR_Fall17_V3_102X_MC_SF_AK8PFchs                                          
SiPixelQualityFromDbRcd      forDigitizer    -                                                                    SiPixelQuality_phase1_2018_v8_mc                                          

Record                   Label         pro::103X_upgrade2018_realistic_HI_v11 Tag  pro::103X_upgrade2018_realistic_HI_Candidate_2019_02_08_17_01_51 Tag  
-----------------------  ------------  ------------------------------------------  --------------------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                           SiPixelQuality_phase1_2018_v10                                        

Record                       Label           pro::103X_upgrade2018_realistic_HEfail_v4 Tag                        pro::103X_upgrade2018_realistic_HEfail_Candidate_2019_02_08_16_57_54 Tag  
---------------------------  --------------  -------------------------------------------------------------------  ------------------------------------------------------------------------  
JetCorrectionsRecord         AK4Calo         JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4Calo     JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4Calo     
JetCorrectionsRecord         AK4JPT          JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4JPT      JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4JPT      
JetCorrectionsRecord         AK4PF           JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PF       JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PF       
JetCorrectionsRecord         AK4PFPuppi      JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PFPuppi  JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PFPuppi  
JetCorrectionsRecord         AK4PFchs        JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK4PFchs    JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK4PFchs    
JetCorrectionsRecord         AK8PF           JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PF       JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PF       
JetCorrectionsRecord         AK8PFPuppi      JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PFPuppi  JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PFPuppi  
JetCorrectionsRecord         AK8PFchs        JetCorrectorParametersCollection_Fall17_17Nov2017_V23_MC_AK8PFchs    JetCorrectorParametersCollection_Fall17_17Nov2017_V32_102X_MC_AK8PFchs    
JetResolutionRcd             AK4PFPuppi_eta  -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PFPuppi                             
JetResolutionRcd             AK4PFPuppi_phi  JR_Spring16_25nsV6_MC_PhiResolution_AK4PFPuppi                       JR_Fall17_V3_102X_MC_PhiResolution_AK4PFPuppi                             
JetResolutionRcd             AK4PFPuppi_pt   JR_Spring16_25nsV6_MC_PtResolution_AK4PFPuppi                        JR_Fall17_V3_102X_MC_PtResolution_AK4PFPuppi                              
JetResolutionRcd             AK4PF_eta       -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PF                                  
JetResolutionRcd             AK4PF_phi       JR_Spring16_25nsV6_MC_PhiResolution_AK4PF                            JR_Fall17_V3_102X_MC_PhiResolution_AK4PF                                  
JetResolutionRcd             AK4PF_pt        JR_Spring16_25nsV6_MC_PtResolution_AK4PF                             JR_Fall17_V3_102X_MC_PtResolution_AK4PF                                   
JetResolutionRcd             AK4PFchs_eta    -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK4PFchs                               
JetResolutionRcd             AK4PFchs_phi    JR_Spring16_25nsV6_MC_PhiResolution_AK4PFchs                         JR_Fall17_V3_102X_MC_PhiResolution_AK4PFchs                               
JetResolutionRcd             AK4PFchs_pt     JR_Spring16_25nsV6_MC_PtResolution_AK4PFchs                          JR_Fall17_V3_102X_MC_PtResolution_AK4PFchs                                
JetResolutionRcd             AK8PFPuppi_eta  -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PFPuppi                             
JetResolutionRcd             AK8PFPuppi_phi  JR_Spring16_25nsV6_MC_PhiResolution_AK8PFPuppi                       JR_Fall17_V3_102X_MC_PhiResolution_AK8PFPuppi                             
JetResolutionRcd             AK8PFPuppi_pt   JR_Spring16_25nsV6_MC_PtResolution_AK8PFPuppi                        JR_Fall17_V3_102X_MC_PtResolution_AK8PFPuppi                              
JetResolutionRcd             AK8PF_eta       -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PF                                  
JetResolutionRcd             AK8PF_phi       JR_Spring16_25nsV6_MC_PhiResolution_AK8PF                            JR_Fall17_V3_102X_MC_PhiResolution_AK8PF                                  
JetResolutionRcd             AK8PF_pt        JR_Spring16_25nsV6_MC_PtResolution_AK8PF                             JR_Fall17_V3_102X_MC_PtResolution_AK8PF                                   
JetResolutionRcd             AK8PFchs_eta    -                                                                    JR_Fall17_V3_102X_MC_EtaResolution_AK8PFchs                               
JetResolutionRcd             AK8PFchs_phi    JR_Spring16_25nsV6_MC_PhiResolution_AK8PFchs                         JR_Fall17_V3_102X_MC_PhiResolution_AK8PFchs                               
JetResolutionRcd             AK8PFchs_pt     JR_Spring16_25nsV6_MC_PtResolution_AK8PFchs                          JR_Fall17_V3_102X_MC_PtResolution_AK8PFchs                                
JetResolutionScaleFactorRcd  AK4PF           JR_Spring16_25nsV6_MC_SF_AK4PF                                       JR_Fall17_V3_102X_MC_SF_AK4PF                                             
JetResolutionScaleFactorRcd  AK4PFPuppi      JR_Spring16_25nsV6_MC_SF_AK4PFPuppi                                  JR_Fall17_V3_102X_MC_SF_AK4PFPuppi                                        
JetResolutionScaleFactorRcd  AK4PFchs        JR_Spring16_25nsV6_MC_SF_AK4PFchs                                    JR_Fall17_V3_102X_MC_SF_AK4PFchs                                          
JetResolutionScaleFactorRcd  AK8PF           JR_Spring16_25nsV6_MC_SF_AK8PF                                       JR_Fall17_V3_102X_MC_SF_AK8PF                                             
JetResolutionScaleFactorRcd  AK8PFPuppi      JR_Spring16_25nsV6_MC_SF_AK8PFPuppi                                  JR_Fall17_V3_102X_MC_SF_AK8PFPuppi                                        
JetResolutionScaleFactorRcd  AK8PFchs        JR_Spring16_25nsV6_MC_SF_AK8PFchs                                    JR_Fall17_V3_102X_MC_SF_AK8PFchs                                          
SiPixelQualityFromDbRcd      forDigitizer    -                                                                    SiPixelQuality_phase1_2018_v8_mc                                          

Record                   Label         pro::103X_upgrade2018cosmics_realistic_deco_v7 Tag  pro::103X_upgrade2018cosmics_realistic_Candidate_2019_02_08_16_59_17 Tag  
-----------------------  ------------  --------------------------------------------------  ------------------------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                                   SiPixelQuality_phase1_2018_v8_mc                                          

[2019-02-11 14:34:22,838] INFO: Connecting to pro [frontier://PromptProd/CMS_CONDITIONS]
Record                   Label         pro::103X_upgrade2018cosmics_realistic_peak_v1 Tag  pro::103X_upgrade2018cosmics_realistic_peak_Candidate_2019_02_08_17_00_21 Tag  
-----------------------  ------------  --------------------------------------------------  -----------------------------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                                   SiPixelQuality_phase1_2018_v8_mc                                               


Record                   Label         pro::103X_postLS2_design_v4 Tag  pro::103X_postLS2_design_Candidate_2019_02_08_17_43_12 Tag  
-----------------------  ------------  -------------------------------  ----------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                SiPixelQuality_phase1_ideal                                 

Record                   Label         pro::103X_postLS2_realistic_v4 Tag         pro::103X_postLS2_realistic_Candidate_2019_02_08_17_03_36 Tag  
-----------------------  ------------  -----------------------------------------  -------------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                          SiPixelQuality_phase1_2017_v10                                 
TrackerAlignmentRcd      -             TrackerAlignment_Upgrade2017_realistic_v2  TrackerAlignment_Upgrade2017_realistic_v3                      


Record                   Label         pro::103X_upgrade2023_realistic_v2 Tag  pro::103X_upgrade2023_realistic_Candidate_2019_02_08_17_01_13 Tag  
-----------------------  ------------  --------------------------------------  -----------------------------------------------------------------  
SiPixelQualityFromDbRcd  forDigitizer  -                                       SiPixelQuality_phase1_2018_v8_mc
```

**Comment by mmusich** [comment]
> DQMHistoTests: Total failures: 1523

given the GT differences reported in https://github.com/cms-sw/cmssw/pull/25885#issuecomment-462333444, it's not surprising one sees differences in the [comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR.html).
Namely one sees changes:
   * in the 2019 workflow [link](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_OldVSNew_TTbar13TeV2019wf11624p0/) due to the change in the Tk Alignment
   * in several miniAOD workflows [1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeV2018wf10824p0), [2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeV2019wf11624p0), [3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeVPUppmx2018wf250202p181) I suppose due to the changes in JECs and JER.

@tocheng is this acceptable or do you wish to get a cleaner comparison by re-making the GTs without spurious differences?

**Comment by tsusa** [comment]
Meanwhile, we are waiting for AlCa to clarify on the Global tags, could simulation and ops people please have a look? @civanch @franzoni @mdhildreth

File: Configuration/Eras/python/Era_Run3_cff.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 from Configuration.Eras.Modifier_run3_HB_cff import run3_HB
 from Configuration.Eras.Modifier_run2_GEM_2017_cff import run2_GEM_2017
 from Configuration.Eras.Modifier_hcalHardcodeConditions_cff import hcalHardcodeConditions
+from Configuration.Eras.Modifier_run2_SiPixel_2018_cff import run2_SiPixel_2018
 
-Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017]), run3_common, run3_GEM, run3_HB, hcalHardcodeConditions)
+
+Run3 = cms.ModifierChain(Run2_2018.copyAndExclude([run2_GEM_2017]), run3_common, run3_GEM, run3_HB, hcalHardcodeConditions, run2_SiPixel_2018)
```
Comment (kpedro88) [inline Configuration/Eras/python/Era_Run3_cff.py]: The `Run3` era starts from `Run2_2018` already. Therefore, `run2_SiPixel_2018` does not need to be added here, since it is already added to `Run2_2018`. Please revert the change to this file.

Comment (tsusa) [reply Configuration/Eras/python/Era_Run3_cff.py]: thanks @kpedro88, fixed by Marco

**Comment by tocheng** [comment]
> > DQMHistoTests: Total failures: 1523
> 
> given the GT differences reported in [#25885 (comment)](https://github.com/cms-sw/cmssw/pull/25885#issuecomment-462333444), it's not surprising one sees differences in the [comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR.html).
> Namely one sees changes:
> 
>     * in the 2019 workflow [link](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_OldVSNew_TTbar13TeV2019wf11624p0/) due to the change in the Tk Alignment
> 
>     * in several miniAOD workflows [1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeV2018wf10824p0), [2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeV2019wf11624p0), [3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_5_X_2019-02-11-1100+25885/30386/validateJR/all_mini_OldVSNew_TTbar13TeVPUppmx2018wf250202p181) I suppose due to the changes in JECs and JER.
> 
> 
> @tocheng is this acceptable or do you wish to get a cleaner comparison by re-making the GTs without spurious differences?

I will clean the spurious change in the GTs and make a new commit.

**Comment by fabiocos** [comment]
@tocheng I assume we need to have your update before another meaningful test may be done. 
@tsusa Concerning the comment of @Martin-Grunewald it looks your feature is activated bu default, could you please clarify?

**Comment by mmusich** [comment]
> @tsusa Concerning the comment of @Martin-Grunewald it looks your feature is activated bu default, could you please clarify?

@fabiocos, allow me to clarify: 
   * in the first push the feature used to be switched off by default in commit https://github.com/cms-sw/cmssw/pull/25885/commits/534bfc3113c1e2f1c5fef8edf5c40b86ea91e42e (because we lacked the appropriate GlobalTags with the right additional label). This was nevertheless giving troubles with the addOnTests, not because of the feature being switched on (indeed in the first round of test the relvals succeeded, see:https://github.com/cms-sw/cmssw/pull/25885#issuecomment-461790255 ), but because in the HLT menus hardocoded in release for the addOn tests there is a use of this module `EventSetupRecordDataGetter` which, I guess, just checks that all the ES products supposed to be produced are indeed available, irrespective if they are consumed or not (after all we are certainly not running the digitizer at HLT). 
   * in the second push (commit https://github.com/cms-sw/cmssw/pull/25885/commits/7fba2f0d940b0a3d639df169c9e79d05132f111e) we updated the PR with Global Tags containing the needed label and switched the feature on: please notice though that since at the moment the labeled and unlabeled version of the `SiPixelQualityRcd` use the same tags, by construction we don't expect any changes in the comparison. The differences spotted in my https://github.com/cms-sw/cmssw/pull/25885#issuecomment-462654058 are due to spurious changes introduced in the GT candidates as the GT queues from which they were derived have evolved in the meantime since the last GTs in autoCond have been cut.

I hope this clarifies. The main point I want to make is that I still fail to understand why this purely technical feature needs to wait upon https://github.com/cms-sw/cmssw/pull/25929

**Comment by davidlange6** [comment]
is it not simply that one wants to avoid supporting candidate GTs in releases?

**Comment by mmusich** [comment]
> is it not simply that one wants to avoid supporting candidate GTs in releases?

fine, but then why not transform candidates in real GTs and move on? It seems one would like to do thorough checks when physics output is changed and technical changes should not be hampered by that (at least that was the spirit when we introduced the candidate concept).

**Comment by davidlange6** [comment]
That sounds good to me.

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
@cmsbuild please test workflow 158.0

**Comment by tocheng** [comment]
+1

**Comment by fabiocos** [comment]
@civanch could you please check and comment or sign? I would like to converge on this before pre2

**Comment by fabiocos** [comment]
+operations

the update of the Era looks coherent with the purpose of the PR to activate the new mechanism for 2018 (and Run2)

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1
